File: core/schemas/qr.py
from datetime import datetime

from pydantic import BaseModel


class QRCodeResponse(BaseModel):
    id: int
    public_code: str
    is_active: bool
    created_at: datetime | None = None
    scan_count: int = 0
    last_scanned_at: datetime | None = None

    class Config:
        from_attributes = True

File: frontend/src/types/qr.ts
export interface QRCodeItem {
  id: number
  public_code: string
  is_active: boolean
  created_at: string | null
  scan_count: number
  last_scanned_at: string | null
}

export interface CreateQRResponse {
  public_code: string
}

export interface DeleteQRResponse {
  message: string
}
